kentikapi/client_integration_test.go

```diff
@@ -518,7 +560,7 @@ func TestClient_GetAgent(t *testing.T) {
 				assert.Equal(t, tt.expectedRequestID, r.GetId())
 			}
 
-			assert.Equal(t, tt.expectedResult.GetAgent(), response.GetAgent())
+			assert.Equal(t, tt.expectedResult.GetAgent().String(), response.GetAgent().String())
```

Comment:
What was the problem with comparing objects?
When there is difference in string representation, there will be no precise diff displayed, so it is inferior solution.

Reply:
On the two occasions that the SpyServer returns valid result this field is different: `state: impl.MessageState{ atomicMessageInfo:(*impl.MessageInfo)(nil)}}`. For some reason when I try to use t.Logf just before this assert the tests pass. It's really weird.

Reply:
`String()` method assures that I'm only comparing the data that's interesting for users and data associated with a given message.

> // The MessageState type holds a *MessageInfo, which must be atomically set to
// the message info associated with a given message instance.

Reply:
Please use proto.Equal for comparing proto objects.

See: https://github.com/stretchr/testify/issues/758#issuecomment-645282184

Reply:
Done; I suggested adding msg in separate comment